Tests/UnitTests/Core/Material/GridSurfaceMaterialTests.cpp

```diff
@@ -223,37 +529,70 @@ BOOST_AUTO_TEST_CASE(GridIndexedMaterial2D) {
   eqeqGrid.atPosition(Point{0.5, std::numbers::pi * 0.75}) = 0u;   // vacuum
 
   // With radius 20
-  auto boundToGrid = std::make_unique<const LocalToZPhi>(20.);
-  Acts::IndexedSurfaceMaterial<EqEqGrid>::BoundToGridLocalDelegate bToZPhi;
-  bToZPhi.connect<&LocalToZPhi::l2ZPhi>(std::move(boundToGrid));
+  auto boundToGridT1 = std::make_unique<const LocalToZPhi>(20.);
+  Acts::IndexedSurfaceMaterial<EqEqGrid>::BoundToGridLocalDelegate bToZPhiT1;
+  bToZPhiT1.connect<&LocalToZPhi::l2ZPhi>(std::move(boundToGridT1));
 
   // With z shift 10
-  auto globalToGrid = std::make_unique<const GlobalToZPhi>(10.);
-  Acts::IndexedSurfaceMaterial<EqEqGrid>::GlobalToGridLocalDelegate gToZphi;
-  gToZphi.connect<&GlobalToZPhi::g2ZPhi>(std::move(globalToGrid));
+  auto globalToGridT1 = std::make_unique<const GlobalToZPhi>(10.);
+  Acts::IndexedSurfaceMaterial<EqEqGrid>::GlobalToGridLocalDelegate gToZphiT1;
+  gToZphiT1.connect<&GlobalToZPhi::g2ZPhi>(std::move(globalToGridT1));
 
   // Create the indexed material grid
   Acts::IndexedSurfaceMaterial<EqEqGrid> ism(
-      std::move(eqeqGrid), Acts::IndexedMaterialAccessor{std::move(material)},
-      std::move(bToZPhi), std::move(gToZphi));
+      std::move(eqeqGrid), Acts::IndexedMaterialAccessor{std::move(materialT1)},
+      std::move(bToZPhiT1), std::move(gToZphiT1));
+
+  // Test with proto grid greation
```

Comment:
For review:

that's the most interesting section probably, it shows how the production of the `GridMaterial` is simplified,
as it will rely only on `ProtoAxes` for the actual grid construction it will remove the template type unrolling in the I/O converters.